backend/lcfs/web/api/fuel_code/repo.py

```diff
@@ -1051,3 +1065,57 @@ async def get_additional_carbon_intensity(
 
         result = await self.db.execute(query)
         return result.scalars().one_or_none()
+
+
+    @repo_handler
+    async def get_default_carbon_intensity(
+        self,
+        fuel_type_id: int,
+        compliance_period: str
+    ) -> Optional[float]:
+        """Get default carbon intensity for specified compliance period"""
+
+        compliance_period_id_subquery = (
+            select(CompliancePeriod.compliance_period_id)
+            .where(CompliancePeriod.description == compliance_period)
+            .scalar_subquery()
+        )
+
+        query = (
+            select(DefaultCarbonIntensity)
+            .where(
+                DefaultCarbonIntensity.fuel_type_id == fuel_type_id,
+                DefaultCarbonIntensity.compliance_period_id == compliance_period_id_subquery
+            )
+        )
+
+        result = await self.db.execute(query)
+        record = result.unique().scalar_one_or_none()
```

Comment:
More unique

Reply:
This is different because it is involving CompliancePeriod and DefaultCarbonIntensity. The relationships between them create eager loading which might create duplicates ( it is the way how fastapi works) and it is required to use unique when you have relationships such as in this case CompliancePeriod and DefaultCarbonIntensity. The other scenario just  adding lazy="selectin" was enough, but in this scenario is recommended to use unique()